Better config, currency support and an amount of doors allow to be owned

RyanTheTechMan opened the pull request:
You can now edit the messages as well as what currency system you plan to use.

New options:
DOOR_CONFIG_DOOR_GROUPS_ENABLED = true -- May not work with door amount at the moment

DOOR_CONFIG_ALLOWED_DOOR_AMOUNT = 3 --How many doors the player may have at any time. If 0, you can own unlimited

Added a way to change the command within the command.

All currencies are supported, but DarkRP, Nut Script, BaseWars, SH Pointshop, & Pointshop are added by default.

THESE CHANGES MOSTLY WORK.
NOT ALL WAS TESTED. PLEASE TEST AND FIX ANYTHING RELATED TO DARK RP!!!

RyanTheTechMan committed:
Better config, currency support and an amount of doors allow to be owned

You can now edit the messages as well as what currency system you plan to use.

New options:
DOOR_CONFIG_DOOR_GROUPS_ENABLED = true -- May not work with door amount at the moment

DOOR_CONFIG_ALLOWED_DOOR_AMOUNT = 3 --How many doors the player may have at any time. If 0, you can own unlimited

Added a way to change the command within the command.

All currencies are supported, but DarkRP, Nut Script, BaseWars, SH Pointshop, & Pointshop are added by default.

THESE CHANGES MOSTLY WORK.
NOT ALL WAS TESTED. PLEASE TEST AND FIX ANYTHING RELATED TO DARK RP!!!

RyanTheTechMan commented:
DOOR_CONFIG_DOOR_GROUPS_ENABLED  may not be needed.

RyanTheTechMan committed:
Made default make amount of doors owned infinite

RyanTheTechMan committed:
Added hooks. Added Billy's Logging support. Fixed some things

LambdaGaming commented on lua/autorun/server/sv_doors.lua:
```diff
@@ -242,7 +265,7 @@ net.Receive( "SyncCoOwner", function( len, sender )
 	local door = ents.GetByIndex( index )
 	if DoorTable[index] then
-		DS_Notify( sender, "This door is managed by a group and cannot be owned." )
+		DS_Notify( sender, "This door is locked ..." )
```
This should be following the same format as the rest of the notifications using the config you made, and it shouldn't be displaying a locked notification since this is used to detect whether or not a group is assigned to the door.
```suggestion
		DS_Notify( sender, DOOR_CONFIG_MESSAGES["Managed By Another Group"] )
```

LambdaGaming reviewed (changes requested):
Aside from the comment I posted, the config that determines what code to run for currency seems to conflict with any gamemodes that don't have a currency system (such as Sandbox). Other than those two things it seems to work great and I really like the changes.

RyanTheTechMan committed:
Update lua/autorun/server/sv_doors.lua

I missed this, thanks!

Co-Authored-By: LambdaGaming <starwars18m@gmail.com>

RyanTheTechMan committed:
Added "Sandbox Currency"

RyanTheTechMan committed:
Moved Allowed Doors to client config

RyanTheTechMan committed:
Added rank management & Door Health

RyanTheTechMan committed:
Defaulted Show Health to false

All of this client config stuff should eventually move to the main config file.

RyanTheTechMan committed:
Added admin bypass to lock other doors on maps

For example, some func doors should be disabled.

RyanTheTechMan committed:
Removed Print. Made offset work
